imagepipeline-base/src/main/java/com/facebook/cache/disk/DefaultDiskStorage.java

```diff
@@ -121,6 +124,24 @@ static String getVersionSubdirectoryName(int version) {
   public boolean isEnabled() {
     return true;
   }
+  
+  @Override
+  public boolean isExternal() {
+    boolean state = false;
```

Comment:
We could compute this only once in the constructor right after setting `mRootDirectory`.

```
mRootDirectory = rootDirectory;
mIsExternal = isExternal(rootDirectory);
```
```
private static boolean isExternal(File directory) {
  ...
}
```